frontend/src/pages/offers/CreateOfferForm.tsx

```diff
@@ -0,0 +1,118 @@
+import { FunctionComponent, ReactNode, useContext, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+import Button from '../../components/Button'
+import ReadOnlyField from '../../components/forms/ReadOnlyField'
+import TextField from '../../components/forms/TextField'
+import { UserProfileContext } from '../../components/UserProfileContext'
+import {
+  GroupType,
+  OfferCreateInput,
+  useAllGroupsMinimalQuery,
+  useShipmentQuery,
+} from '../../types/api-types'
+import { formatShipmentName } from '../../utils/format'
+
+interface Props {
+  /**
+   * This shipment ID to determine where this offer will be created.
+   */
+  shipmentId: number
+  /**
+   * If true, the Submit button will be disabled
+   */
+  isLoading: boolean
+  /**
+   * The text to display in the Submit button of the form
+   */
+  submitButtonLabel: ReactNode
+  /**
+   * The callback triggered when the user submits the form
+   */
+  onSubmit: (input: OfferCreateInput) => void
+}
+
+/**
+ * This form allows users to create a new offer. It is very specific to offer
+ * creation, and therefore difficult to reuse for editing offers.
+ */
+const OfferForm: FunctionComponent<Props> = (props) => {
+  // Each group leader is assigned to a single group
+  // We need to figure out which group that is
+  // TODO: handle the fact that admins can be assigned to any number of groups
```

Comment:
many user accounts per group relationships is something to chat about

Reply:
Isn't it the opposite? Many groups per user account, right?

Reply:
many to many I'm afraid 😱